### PR TITLE
[refactor] make s2p a python package that can be installed with pip

### PR DESCRIPTION
**Pull request by carlodef**
This pull request reorganises the sources in order to make `s2p` installable with `pip`. The main changes are:

* `s2p` is now a python package (instead of a python module)
* the folder `s2plib` was renamed to `s2p`
* the file `s2p.py` has disappeared and its content is split among `s2p/__init__.py` (most of the code) and `s2p/cli.py` (command line parsing code)
* when one installs `s2p` with `pip`, a CLI tool named `s2p` becomes available in the system PATH
* the file `s2p_test.py` was moved to `tests/test_s2p.py`
* the folder `testdata` was moved to `tests/data` 
* the README has been updated with simpler install instructions and CLI usage examples

Python codes that depend on `s2p` should not be affected: `import s2p` still works in the same way.

Shell scripts running `python /path/to/your/clone/of/s2p/s2p.py` should be updated to run instead `s2p` as a standalone command line tool.

This pull request contains the changes of pull request #191. 

EDIT: A source distribution of `s2p` corresponding to this branch's version is available on [PyPI](https://pypi.org/) and can be installed with `pip`: 

    pip install s2p

### CODE REVIEW

**Comment by jmichel-otb** [comment]
I think there should be only a few minor conflicts with #196:
* Name of the data dir
* Imports (unlikely)
* Calls to read_config_file and main in s2p.py

File: s2p/__init__.py [hunk 1]
```diff
@@ -33,17 +33,17 @@
...
+from s2p import pointing_accuracy
+from s2p import rectification
+from s2p import block_matching
+from s2p import masking
+from s2p import triangulation
+from s2p import fusion
+from s2p import visualisation
 
 
 def pointing_correction(tile, i):
```
Comment (jmichel-otb) [inline s2p/__init__.py:49]: Maybe those functions should not be in init but in a dedicated module in s2p package no ? If we were to write tests for each of them, how would we proceed ?

**Review by jmichel-otb — APPROVED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -25,14 +25,13 @@ before_install:
   - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew unlink gdal && (brew install gcc || brew link --overwrite gcc) && brew install geographiclib fftw; fi
 
 install:
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then pip install numpy utm bs4 requests pyproj; fi
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then pip install rasterio; fi
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then pip install numpy; fi
```
Comment (jmichel-otb) [inline .travis.yml:28]: Can you explain why you need to keep numpy installation here ? Other packages seems to be handled by `config.py`.

Comment (carlodef) [reply .travis.yml:28]: > Other packages seems to be handled by `config.py`

I suppose you mean `setup.py`?

I agree with you that this is weird, but numpy is required to install `rasterio`. Quoting https://rasterio.readthedocs.io/en/stable/installation.html#installing-from-the-source-distribution:

> the Numpy headers are required to run the rasterio setup script. Numpy has to be installed (via the indicated requirements file) before rasterio can be installed.

In rasterio's `setup.py` source code, we can see that `numpy` is required:
https://github.com/mapbox/rasterio/blob/master/setup.py#L86

Unfortunately having `numpy`  in the list of our dependencies in `setup.py` is not enough for `rasterio`, and `numpy` has to be installed separately beforehand.

**Comment by jmichel-otb** [comment]
I just did a merge by hand with #196 and it is fairly easy. There are a few minor conflicts on `.travis.yaml` and imports needs to be changed from `s2plib` to `s2p` in `tests` folder.

**Pull request merged**